src/pyramid/security.py

```diff
@@ -280,86 +293,155 @@ class ACLAllowed(ACLPermitsResult, Allowed):
     """
 
 
-class AuthenticationAPIMixin(object):
+class SecurityAPIMixin(object):
     @property
-    def authenticated_userid(self):
-        """ Return the userid of the currently authenticated user or
-        ``None`` if there is no :term:`authentication policy` in effect or
-        there is no currently authenticated user.
+    def authenticated_identity(self):
+        """
+        Return an opaque object identifying the current user or ``None`` if no
+        user is authenticated or there is no :term:`security policy` in effect.
 
-        .. versionadded:: 1.5
         """
-        policy = _get_authentication_policy(self)
+        policy = _get_security_policy(self)
         if policy is None:
             return None
-        return policy.authenticated_userid(self)
+        return policy.identify(self)
 
+    @property
+    def authenticated_userid(self):
+        """
+        Return the :term:`userid` of the currently authenticated user or
+        ``None`` if there is no :term:`security policy` in effect or there is
+        no currently authenticated user.
+
+        .. versionchanged:: 2.0
+
+            When using the new security system, this property outputs the
+            string representation of the :term:`identity`.
+
+        """
+        authn = _get_authentication_policy(self)
+        security = _get_security_policy(self)
+        if authn is not None:
+            return authn.authenticated_userid(self)
+        elif security is not None:
+            return str(security.identify(self))
```

Comment:
@luhn can I ask why this method checks old-style authn policy before security policy?
The other methods in the mixin only work with security policy, without falling back on (or checking first!) an authentication policy.

Is there a goal to support a config with security policy + authentication policy?

Reply:
Usually we just let LegacySecurityPolicy handle the bw compat, but in this case we need to circumvent the `str(security.identify(self))`, otherwise we'd be changing the behavior of the legacy method.

Now that `str(identity)` is out and authenticated_userid is back, we should be able to just reference the security policy and let LegacySecurityPolicy work its magic.

Reply:
Great! :sparkles: